plugins/node/instrumentation-undici/src/internal-types.ts

```diff
@@ -13,14 +13,21 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-import type { Channel } from 'diagnostics_channel';
+
+import type { Channel, ChannelListener } from 'diagnostics_channel';
 
 import { UndiciRequest, UndiciResponse } from './types';
 
+export interface DiagnosticChannelApi {
```

Comment:
Perhaps "DiagnosticsChannelApi" (i.e. plural "Diagnostics" to match `diagnostics_channel`).